lib/game_engine.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'user_interface'
+class GameEngine
+  include UserInterface
+  attr_reader :playing_board, :player1, :player2
+
+  def initialize(board, player1, player2)
+    @playing_board = board
+    @player1 = player1
+    @player2 = player2
+  end
+
+  def play
+    current_player = player1
+    loop do
+      UserInterface.display_which_player_turn(current_player)
+      move = current_player.make_move(playing_board)
+      next if move.nil?
+
+      playing_board.apply_move(current_player.symbol, move)
+
+      return find_board_status(current_player) unless find_board_status(current_player).nil?
+
+      current_player = switch_players(current_player)
+    end
```

Comment:
It might be good to print out the game result rather than returning it to the caller.

Reply:
Resolved at 2e23d421ffcb2a650b6b6fa8f392e0b8989f504f